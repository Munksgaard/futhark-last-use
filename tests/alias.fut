let main (x: i32): i32 =
  let xs = iota x
  let xs[1] = xs[0]

  let ys =
    if xs[0] > 0 then
      xs
    else
      indices xs

  let y = reduce (+) 0 ys

  in y
