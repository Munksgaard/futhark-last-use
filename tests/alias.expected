Up to date
(valid_24, [bounds_invalid_upwards_23])
(range_valid_c_25, [valid_24])
(index_certs_28, [y_27])
(index_certs_30, [y_29])
(xs_31, [res_26])
(cond_33, [x_32])
(ys_34, [cond_33])
(res_36, [res_48])
(res_39, [x_37, x_38])
(x_40, [xs_31, ys_34, gtid_50])
(x_41, [x_22])
(num_groups_46, [x_41])
(res_48, [segred_group_size_44, num_groups_46])
(bytes_53, [binop_x_54])
(mem_55, [bytes_53])
(bytes_56, [binop_x_57])
(mem_58, [bytes_56])

